File: python/run_model.py
"""Python Model Example"""

import os

import numpy as np
import pandas as pd
import typer
from typing_extensions import Annotated


def predict(df: pd.DataFrame) -> pd.DataFrame:
    """Sample prediction function.

    TODO: Replace this with your actual model prediction logic. In this
    example, random floats are assigned.
    """
    pred = df.loc[:, ["id"]]
    pred["probability"] = np.random.random_sample(size=len(pred.index))
    return pred


def main(
    input_dir: Annotated[str, typer.Option()] = "/input",
    output_dir: Annotated[str, typer.Option()] = "/output",
):
    """
    Run inference using data in input_dir and output predictions to output_dir.
    """
    data = pd.read_csv(os.path.join(input_dir, "data.csv"))
    predictions = predict(data)
    predictions.to_csv(
        os.path.join(output_dir, "predictions.csv"),
        index=False,
    )


if __name__ == "__main__":
    typer.run(main)

File: r/run_model.R
# R Model Example

library(optparse)
library(readr)
library(dplyr)

predict <- function(df) {
  #' Sample prediction function.
  #'
  #' TODO: Replace this with your actual model prediction logic. In this
  #' example, random floats are assigned.
  #'
  #' @param df A data frame.
  #' @return A data frame with predictions.

  pred <- df %>% select(id)
  pred$probability <- runif(nrow(pred))
  pred
}

# Create a CLI ------------------------------------
# Two args: `input-dir`, `output-dir`
option_list <- list(
  make_option("--input-dir", type="character", default="/input", metavar="character"),
  make_option("--output-dir", type="character", default="/output", metavar="character")
)
opt_parser <- OptionParser(option_list=option_list)
opt <- parse_args(opt_parser)


# Read in input file ------------------------------
data <- read_csv(file.path(opt[["input-dir"]], "data.csv"), show_col_types = FALSE)


# Run a prediction --------------------------------
# Randomly generate a float between [0.0, 1.0)
predictions <- predict(data)


# Output predictions to output file ---------------
write_csv(predictions, file.path(opt[["output-dir"]], "predictions.csv"))
